javascript/src/refactor.js

```diff
@@ -498,6 +524,64 @@ class RefactorEngine {
                         return this.remove();
                     }
                 } else if (node.type === 'Identifier') {
+                    // this is the part where the function arguments are substituted values if they are a part of the assignments
+                    if (node.name in assignments) {
+                        if (assignments[node.name] === true) {
+                            engine.changed = true;
+                            return engine.trueLiteral();
+                        } else if (assignments[node.name] === false) {
+                            engine.changed = true;
```

Comment:
should `engine.changed` be false here? if not, can we pull out the two assignments outside the `if statement`

Reply:
```engine.changed``` must be true here. I'll refine the code such that it isnt redundant